src/main/java/NetworkView.java

```diff
@@ -0,0 +1,89 @@
+import javax.swing.*;
+import java.awt.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.HashMap;
+
+public class NetworkView extends JComponent {
+
+    private final Graph graph;
+    private int viewSize;
+    int nrNodes;
+    int[] xyValues = new int[2];
+
+    List<GraphNode> listNodes = new ArrayList<>();
+
+    HashMap<GraphNode, int[]> nodeValues = new HashMap<>();
+
+    public NetworkView(Graph graph) {
+        this.graph = graph;
+    }
+
+    public int getViewSize() {
+        return viewSize;
+    }
+
+    @Override
+    protected void paintComponent(Graphics g) {
+        super.paintComponent(g);
+        paintNodes(g);
+        paintConnections(g);
+    }
+
+    void paintNodes(Graphics g) {
+        listNodes = graph.getGraph();
+        nrNodes = graph.getGraph().size();
+        int networkRadius = nrNodes * 10;
+        viewSize = networkRadius*2 + 300;
```

Comment:
I want to update the view's size based on the network radius which is based on the number of nodes, but it only uses the value initialized outside the method.